### ISSUE TITLE
devpi list and devpi remove no longer consider '.dev' versions

### ISSUE THREAD

**Issue by wouterdb** [devpi/devpi#928]
# Problem

`devpi list $package` and  `devpi remove $package` no longer consider dev versions

The problem does not exist on `devpi-common==3.6.0` but it does on `devpi-common==3.7.0`

# Example (devpi-client==6.0.2, devpi-common==3.7.0)

The cli reports no dev versions
```
python -m devpi --debug remove -vvv "inmanta-core<7.1.1.dev20221019122458,>7.1.1.dev0"
No releases or distributions found matching 'inmanta-core<7.1.1.dev20221019122458,>7.1.1.dev0'.

python -m devpi --debug list inmanta-core                                                       
https://**/root/pypi/+f/279/333b0b015ba40/inmanta-core-7.1.0.tar.gz
https://**/root/pypi/+f/c7e/f03485b7d6979/inmanta_core-7.1.0-py3-none-any.whl
https://**/root/pypi/+f/7f6/81201a058fe80/inmanta-core-7.0.0.tar.gz
....
```
The web ui does
![image](https://user-images.githubusercontent.com/1788254/196701732-5bb4fc46-65db-48ca-ba20-fac5088698f6.png)

If I slightly change the filter, it does find them

```
python -m devpi --debug list "inmanta-core>=0.0.0.dev0"      
https://**/inmanta/dev/+f/b44/7a945cf99c3c9/inmanta-core-7.1.1.dev20221019122458.tar.gz
https://**/inmanta/dev/+f/e4b/e2f115321a892/inmanta_core-7.1.1.dev20221019122458-py3-none-any.whl
https://**/inmanta/dev/+f/065/8aba873c13fc6/inmanta-core-7.1.1.dev20221019121532.tar.gz
https://**/inmanta/dev/+f/2bc/c7658f2359d34/inmanta_core-7.1.1.dev20221019121532-py3-none-any.whl
https://**/inmanta/dev/+f/dd0/0d23f9107b599/inmanta-core-7.1.1.dev20221019035553.tar.gz
https://**/inmanta/dev/+f/cc2/2c22bb8ebfbe4/inmanta_core-7.1.1.dev20221019035553-py3-none-any.whl
```

However, not when I add the upper bound

```
 python -m devpi list "inmanta-core<7.1.1.dev20221019122458"
https://**/root/pypi/+f/279/333b0b015ba40/inmanta-core-7.1.0.tar.gz
https://**/root/pypi/+f/c7e/f03485b7d6979/inmanta_core-7.1.0-py3-none-any.whl
https://**/root/pypi/+f/7f6/81201a058fe80/inmanta-core-7.0.0.tar.gz
https://**/root/pypi/+f/bd6/ac4706bf7de8d/inmanta_core-7.0.0-py3-none-any.whl
https://**/root/pypi/+f/423/6188b1117f650/inmanta-core-6.3.0.tar.gz
https://**/root/pypi/+f/256/3c78698b4de3a/inmanta_core-6.3.0-py3-none-any.whl
``` 

# Root cause 

I have this script to clean out dev artifacts, which works on `5.2.2` and not on `6.0.2` 

```bash
#!/bin/bash

devpi use https://**/inmanta/dev

# Get all packages names
devpi list >packages

# put in array
mapfile -t packages <packages

for P in ${packages[@]}; do
   echo "begin $P"
   # get every file we have, find the dev version
   devpi list $P --all| egrep -o "([0-9]+[.]){1,2}([0-9]+)[.]?dev[0-9]+">versions_for
   # get all the version before dev
   cat versions_for | egrep -o "([0-9]+[.]){1,2}([0-9]+)" | sort | uniq>majors
   mapfile -t majors <majors
   for M in ${majors[@]}; do
      # for every version, keep only the lastest dev version
      final_dev=$(grep ^$M versions_for | sort | tail -1)
      devpi remove -y "$P<$final_dev,>$M.dev"
   done
done
``` 

# Info 

- I'm on 6.0.2
- Python 3.10.7, fedora 36

## Pip list
```
Package              Version
-------------------- -----------
aiohttp              3.8.3
aiosignal            1.2.0
argon2-cffi          21.3.0
argon2-cffi-bindings 21.2.0
async-timeout        4.0.2
attrs                22.1.0
beautifulsoup4       4.11.1
bleach               5.0.1
build                0.8.0
certifi              2022.9.24
cffi                 1.15.1
Chameleon            3.10.1
charset-normalizer   2.1.1
check-manifest       0.48
cmarkgfm             0.8.0
defusedxml           0.7.1
devpi                2.2.0
devpi-client         6.0.2
devpi-common         3.7.0
devpi-server         6.7.0
devpi-web            4.1.1
docutils             0.19
frozenlist           1.3.1
hupper               1.10.3
idna                 3.4
itsdangerous         2.1.2
lazy                 1.5
multidict            6.0.2
packaging            21.3
passlib              1.7.4
PasteDeploy          3.0.1
pep517               0.13.0
pip                  22.3
pkginfo              1.8.3
plaster              1.0
plaster-pastedeploy  0.7
platformdirs         2.5.2
pluggy               1.0.0
py                   1.11.0
pycparser            2.21
Pygments             2.13.0
pyparsing            3.0.9
pyramid              2.0
pyramid-chameleon    0.3
python-dateutil      2.8.2
readme-renderer      37.2
repoze.lru           0.7
requests             2.28.1
ruamel.yaml          0.17.21
ruamel.yaml.clib     0.2.6
setuptools           65.3.0
six                  1.16.0
soupsieve            2.3.2.post1
strictyaml           1.6.2
tomli                2.0.1
translationstring    1.4
urllib3              1.26.12
venusian             3.0.0
waitress             2.1.2
webencodings         0.5.1
WebOb                1.8.7
wheel                0.37.1
Whoosh               2.7.4
yarl                 1.8.1
zope.deprecation     4.4.0
zope.interface       5.5.0
```

**Comment by fschulze** [comment]
This broke with the switch from ``pkg_resources`` to ``packaging``. Could you try the fix from this commit: https://github.com/fschulze/devpi/commit/fb5fa6583a8cc0ae16cb5ad1671b2402beec2b92 ?

**Comment by fschulze** [comment]
Should be fixed with devpi-common 3.7.2.
